src/pages-and-resources/discussions/app-config-form/apps/shared/DivisionByGroupFields.jsx

```diff
@@ -1,61 +1,142 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
 import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 import { Form, TransitionReplace } from '@edx/paragon';
+import { FieldArray, useFormikContext } from 'formik';
 import FormSwitchGroup from '../../../../../generic/FormSwitchGroup';
 import messages from './messages';
 import AppConfigFormDivider from './AppConfigFormDivider';
+import { updateDividedDiscussionsIds } from '../../../data/slice';
+import { updateModel } from '../../../../../generic/model-store';
 
 function DivisionByGroupFields({
-  onBlur,
-  onChange,
-  intl,
-  values,
+  onBlur, onChange, intl, appConfig,
 }) {
+  const dispatch = useDispatch();
+  const { setFieldValue } = useFormikContext();
+  const {
+    dividedDiscussionsIds,
+    discussionTopics,
+    divideByCohorts,
+    divideCourseTopics,
+  } = appConfig;
+
+  useEffect(() => {
+    dispatch(updateDividedDiscussionsIds({ dividedDiscussionsIds }));
+  }, [dividedDiscussionsIds]);
+
+  useEffect(() => {
+    const discussionTopicIds = discussionTopics.map(
+      (topic) => topic.id,
+    );
+    const divideCourseTopicsSwitchOff = (
+      discussionTopicIds.length === dividedDiscussionsIds.length
+      && discussionTopicIds.every((topicId) => dividedDiscussionsIds.includes(topicId))
+    ) || !dividedDiscussionsIds.length;
+
+    if (divideByCohorts) {
+      if (divideCourseTopicsSwitchOff && !divideCourseTopics) {
+        setFieldValue('divideCourseTopics', false);
+        setFieldValue('dividedDiscussionsIds', discussionTopicIds);
+      } else {
+        setFieldValue('divideCourseTopics', true);
+      }
+    } else {
+      setFieldValue('dividedDiscussionsIds', []);
+      setFieldValue('divideCourseTopics', false);
+    }
+
+    const {
+      dividedDiscussionsIds: courseDividedDiscussionsIds,
+      discussionTopics: courseDiscussionTopics,
+      ...payload
+    } = appConfig;
+    dispatch(updateModel({ modelType: 'appConfigs', model: payload }));
+  }, [
+    divideByCohorts,
+    divideCourseTopics,
+    discussionTopics,
+  ]);
+
+  const handleCheckBoxToggle = (event, push, remove) => {
+    const { checked, value } = event.target;
+    if (checked) {
+      push(value);
+    } else {
+      remove(dividedDiscussionsIds.indexOf(value));
+    }
+  };
+
+  const handleDivideCourseWideTopicsSwitch = (event) => {
+    const { checked } = event.target;
+    if (!checked) {
+      setFieldValue('dividedDiscussionsIds', []);
+    }
+    onChange(event);
```

Comment:
What do we need to call this onChange?

Reply:
this onChange updates the toggle value of divide Course  wide Topics switch.